### ISSUE TITLE
greenkeeper-lockfile-upload cannot push when use GK_LOCK_COMMIT_AMEND

### ISSUE THREAD

**Issue by guillaumearm** [greenkeeperio/greenkeeper-lockfile#152]
Since #126, we can set `GK_LOCK_COMMIT_AMEND` to a truthy value to `git commit --amend --no-edit` the "update lockfile" commit.

However, `greenkeeper-lockfile-upload` cannot push because HEAD diverged from FETCH_HEAD.

2 solutions : 
- add a `GK_LOCK_PUSH_FORCE` environment flag
or
- `push --force` should be used when `GK_LOCK_COMMIT_AMEND` is truthy

**Comment by finnp** [comment]
Good catch. I think using `--force-with-lease` when `GK_LOCK_COMMIT_AMEND` specified would be the way to go.
